File: Alkitab/src/yuku/alkitab/base/model/SongInfo.java
package yuku.alkitab.base.model;

import yuku.kpri.model.Song;


/** A concise record of a song. The field song may not be filled if it's not yet loaded */
public class SongInfo {
	public String bookName;
	public String code;
	public String title;
	public String title_original;
	public Song song;
	
	public SongInfo(String bookName, String code, String title, String title_original) {
		this.bookName = bookName;
		this.code = code;
		this.title = title;
		this.title_original = title_original;
	}
}

File: Alkitab/src/main/java/yuku/alkitab/songs/SongInfo.kt
package yuku.alkitab.songs

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * A concise record of a song without "opening" its contents.
 *
 * [snippet] is populated only by the deep ("include lyrics") search path
 * ([yuku.alkitab.base.storage.SongDb.listSongInfosByBookNameAndDeepFilter]) and
 * holds up to a couple of matching lyric lines to preview under the search result.
 * It is null for ordinary metadata listings.
 */
@Parcelize
class SongInfo @JvmOverloads constructor(
    @JvmField
    val bookName: String,
    @JvmField
    val code: String,
    @JvmField
    val title: String,
    @JvmField
    val title_original: String?,
    @JvmField
    val snippet: String? = null,
) : Parcelable
